### ISSUE TITLE
Publish a docker image for Delta Lake

### ISSUE THREAD

**Issue by zsxwing** [delta-io/delta#919]
Currently it's not convenient to try out Delta Lake. People need to install Spark first, and following multiple steps in https://docs.delta.io/latest/quick-start.html#set-up-apache-spark-with-delta-lake

It would be great if we can publish a docker image for Delta Lake so that people can try it out using a simple docker command.

For such docker image, we can maintain a Dockerfile in the repo and publish a new image to https://hub.docker.com/u/deltaio in each release.

**Comment by RishiKumarRay** [comment]
Do we have to write a dockerfile?

**Comment by zsxwing** [comment]
@RishiKumarRay I think it's better to maintain a dockerfile in GitHub and we build and publish a new image when making a new release. Do you have any other suggestion?

**Comment by RishiKumarRay** [comment]
@zsxwing  yeah i think its great to maintain a dockerfile for it on github itself, can you tell me steps for building this?

**Comment by RishiKumarRay** [comment]
so we need a github actions to publish images

**Comment by zsxwing** [comment]
> so we need a github actions to publish images

No. We don't need a github actions. Currently, our release process is manual. We will build the image from the dockerfile manually when making a new release.

> what is the intended entrypoint for the docker image? the various different shell options?

Yep. I think the default can be `pyspark`. We can also provide options for spark-shell, or even a jupyter notebook.

**Comment by mdrakiburrahman** [comment]
@zsxwing @RishiKumarRay - any updates on this issue? 

I'm looking for a simple Java (non-Spark) Docker Container that let's anyone reproduce the ["Zappy" fake example in the Delta docs](https://docs.delta.io/latest/delta-standalone.html#mock-situation) and in [this Databricks blog](https://databricks.com/blog/2022/01/28/the-ubiquity-of-delta-standalone-java-scala-hive-presto-trino-power-bi-and-more.html) - in a simple "Hello World" manner without installing anything locally.

It's not easy for someone coming into the Delta ecosystem to get up to speed - having a reproducible example managed as a Dockerfile in this repo would be a phenomenal value add. 

I see the PR by @stikkireddy - I don't think this meets the use case, because he's using Spark in his [Dockerfile](https://github.com/delta-io/delta/blob/e35bd9c6098d5015ec5b1d226ad4b9e24b86dc0b/examples/Dockerfile#L6) - and its just a wrapper around Jupyter notebooks (i.e. a simplified Databricks). 

Rather, there should be a vanilla Java Dockerfile that has nothing to do with Apache Spark to uphold the theme of the Standalone writer - let non-Spark clients become a first class Delta citizen. It's difficult for devs to do this if the barrier to entry is high, a Dockerfile would reduce this to 5 mins.

It should be trivial to create such a Dockerfile for the maintainers of this repo, I'm really surprised it hasn't been done yet.

**Comment by allenhaozi** [comment]
We will run spark job base on delta lake on top of K8S and also need such a Dockerfile

**Comment by avantgardnerio** [comment]
@zsxwing is this what you were looking for? https://github.com/spaceandtimelabs/docker-spark-deltalake

**Comment by MrPowers** [comment]
@zsxwing - looks like there is [already a PR open for this issue](https://github.com/delta-io/delta/pull/922).  What are your recommended next steps?  If a new contributor would like to work on this issue, should they just open another PR?

**Comment by zsxwing** [comment]
Feel free to leave a comment in the open PR and pick up whatever is left.

**Comment by dennyglee** [comment]
I'll comment directly in PR #922 but for a Spark-based Docker, I'm wondering if it would be helpful if we leveraged the proposed Spark docker per [SPIP: Support Docker Official Image for Spark](https://docs.google.com/document/d/1nN-pKuvt-amUcrkTvYAQ-bJBgtsWb9nAkNoVNRM2S2o/edit#heading=h.nqyzx5gvu8rl).

**Comment by alberttwong** [comment]
https://github.com/delta-io/delta-docs/tree/main/static/quickstart_docker

**Comment by dennyglee** [comment]
Thanks @alberttwong  - completely spaced out on this!

Yes, we have both the docker code as you noted as well as we pushed the Delta docker to DockerHub at http://go.delta.io/dockerhub.  

I'm thinking that we need to create a separate docker repo (e.g., https://github.com/delta-io/docker) so we can automate the docker builds.    Saying this, as the docker has been created, thinking this may be appropriate to close.  Will leave it open for a short while for more comments before closing it, eh?!

**Comment by foss-contributor** [comment]
@zsxwing, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
